tools/pnnx/cmake/PNNXPyTorch.cmake

```
@@ -0,0 +1,67 @@
+# reference to https://github.com/llvm/torch-mlir/blob/main/python/torch_mlir/dialects/torch/importer/jit_ir/cmake/modules/TorchMLIRPyTorch.cmake
+
+# PNNXProbeForPyTorchInstall
+# Attempts to find a Torch installation and set the Torch_ROOT variable
+# based on introspecting the python environment. This allows a subsequent
+# call to find_package(Torch) to work.
+function(PNNXProbeForPyTorchInstall)
+  if(Torch_ROOT)
+    message(STATUS "Using cached Torch root = ${Torch_ROOT}")
+  elseif(Torch_INSTALL_DIR)
+    message(STATUS "Using cached Torch install dir = ${Torch_INSTALL_DIR}")
+    set(Torch_DIR "${Torch_INSTALL_DIR}/share/cmake/Torch" CACHE STRING "Torch dir" FORCE)
+  else()
+    #find_package (Python3 COMPONENTS Interpreter Development)
+    find_package (Python3)
+    message(STATUS "Checking for PyTorch using ${Python3_EXECUTABLE} ...")
+    execute_process(
+      COMMAND "${Python3_EXECUTABLE}"
+      -c "import os;import torch;print(torch.utils.cmake_prefix_path, end='')"
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      RESULT_VARIABLE PYTORCH_STATUS
+      OUTPUT_VARIABLE PYTORCH_PACKAGE_DIR)
+    if(NOT PYTORCH_STATUS EQUAL "0")
+      message(STATUS "Unable to 'import torch' with ${Python3_EXECUTABLE} (fallback to explicit config)")
+      return()
+    endif()
+    message(STATUS "Found PyTorch installation at ${PYTORCH_PACKAGE_DIR}")
+
+    set(Torch_ROOT "${PYTORCH_PACKAGE_DIR}" CACHE STRING
+        "Torch configure directory" FORCE)
+  endif()
+endfunction()
+
+# PNNXConfigurePyTorch
+# Performs configuration of PyTorch flags after CMake has found it to be
+# present. Most of this comes down to detecting whether building against a
+# source or official binary and adjusting compiler options in the latter case
+# (in the former, we assume that it was built with system defaults). We do this
+# conservatively and assume non-binary builds by default.
+#
+# In the future, we may want to switch away from custom building these
+# extensions and instead rely on the Torch machinery directly (definitely want
+# to do that for official builds).
+function(PNNXConfigurePyTorch)
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    # Linux specific libstdcpp ABI checking.
+    message(STATUS "Checking if Torch is an official binary ...")
+    execute_process(
+      COMMAND ${Python3_EXECUTABLE}
+      -c "from torch.utils import cpp_extension as c; import sys; sys.exit(0 if c._is_binary_build() else 1)"
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      RESULT_VARIABLE _is_binary_build)
+    if(${_is_binary_build} EQUAL 0)
+      set(TORCH_CXXFLAGS "")
+      if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+        set(TORCH_CXXFLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11")
+      elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
```

```suggestion
      elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```
写成这样会比较好。如果不这样的话，遇到 CMAKE_CXX_COMPILER_ID 的值为空的极端情况，这行 cmake 代码会变成 `elseif( STREQUAL "Clang")` 然后报语法错误

---

已修改